boot/initramfs_test.go

```diff
@@ -630,3 +631,114 @@ func (s *initramfsSuite) TestInitramfsRunModeSelectSnapsToMount(c *C) {
 		}
 	}
 }
+
+func (s *initramfsSuite) TestInitramfsRunModeUpdateBootloaderVars(c *C) {
+	bloader := bootloadertest.Mock("noscripts", c.MkDir()).WithNotScriptable()
+	bootloader.Force(bloader)
+	defer bootloader.Force(nil)
+
+	tt := []struct {
+		cmdline       string
+		initialStatus string
+		finalStatus   string
+	}{
+		{
+			cmdline:       "kernel_status=trying",
+			initialStatus: "try",
+			finalStatus:   "trying",
+		},
+		{
+			cmdline:       "kernel_status=trying",
+			initialStatus: "badstate",
+			finalStatus:   "",
+		},
+		{
+			cmdline:       "kernel_status=trying",
+			initialStatus: "",
+			finalStatus:   "",
+		},
+		{
+			cmdline:       "",
+			initialStatus: "try",
```

Comment:
can we have a test case with `initialStatus` of trying too? that should end up being cleared in the case where the boot was not successful (from the standpoint of userspace eventually reading the finalStatus at least)